### PR TITLE
dpdk: added VERIFY-DPDK-NFF-GO TC

### PR DESCRIPTION
**Pull request by ader1990**
VERIFY-DPDK-NFF-GO test case:

  * Build DPDK
  * Build NFF-GO using the previously build DPDK
  * Run 'make citesting' inside nff-go directory -- which will run a suite of nff-go tests which use the MLX PMD

### CODE REVIEW

File: Testscripts/Linux/build_test_nff_go.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	SetTestStateRunning
+	LogMsg "Configuring ${1} ${DISTRO_NAME} ${DISTRO_VERSION} for NFF-GO test..."
+	packages=(make git curl wget libpcap-dev libelf-dev hugepages libnuma-dev libhyperscan-dev liblua5.3-dev libmnl-dev libibverbs-dev)
+	case "${DISTRO_NAME}" in
+		ubuntu|debian)
+			ssh "${1}" "until dpkg --force-all --configure -a; sleep 10; do echo 'Trying again...'; done"
+			ssh "${1}" ". ${UTIL_FILE} && update_repos"
+			;;
+		*)
+			echo "Unknown distribution"
```
Comment (chvalean) [inline Testscripts/Linux/build_test_nff_go.sh]: Unsupported distribution.

File: Testscripts/Linux/build_test_nff_go.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+			SetTestStateAborted
```
Comment (chvalean) [inline Testscripts/Linux/build_test_nff_go.sh]: Test Skip maybe?

File: Testscripts/Linux/build_test_nff_go.sh [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	ssh "${1}" "cd ${NFF_GO_DIR}/examples && eval '${exported_vars} make -j nffPktgen'"
+	ssh "${1}" "cd ${NFF_GO_DIR} && eval '${exported_vars} make -j -C examples/dpi'"
+
+	# Start NFF-GO tests
+	ssh "${1}" "cd ${NFF_GO_DIR} && eval '${exported_vars} make citesting'"
+	check_exit_status "make citesting on ${1}" "exit"
+	ssh "${1}" "cd ${NFF_GO_DIR} && eval '${exported_vars} make testing'"
+	check_exit_status "make testing on ${1}" "exit"
+
+	LogMsg "Built and rand tests for NFF-GO on ${1}"
```
Comment (chvalean) [inline Testscripts/Linux/build_test_nff_go.sh]: ran*

File: Testscripts/Windows/VERIFY-DPDK-NFF-GO.ps1 [hunk 4]
```diff
@@ -0,0 +1,145 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+param([object] $AllVmData,
+		[object] $CurrentTestData)
+
+function Get-TestStatus {
```
Comment (chvalean) [inline Testscripts/Windows/VERIFY-DPDK-NFF-GO.ps1]: can this be put to some of the main libraries as a generic function?

**Comment by ader1990** [comment]
Execution tested:

```
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-DAILY-LTS : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:42

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 DPDK                 VERIFY-DPDK-NFF-GO                                                                PASS                26.21
```

**Review by chvalean — APPROVED** [review]

**Pull request merged**